File: server/in-memory-database/src/main/java/uk/co/stevewellington/shoppinglist/infrasctructure/InMemoryItemStore.java
package uk.co.stevewellington.shoppinglist.infrasctructure;

import com.google.common.collect.Sets;
import uk.co.stevewellington.shoppinglist.model.ItemName;

import java.util.Set;

import static com.google.common.collect.Sets.newHashSet;

public class InMemoryItemStore implements ItemStore {

    private final Set<ItemName> itemNames = newHashSet();

    @Override
    public void createItemName(ItemName itemName) {
        itemNames.add(itemName);
    }
}

File: server/in-memory-database/src/main/scala/uk/co/stevewellington/shoppinglist/InMemoryItemStore.scala
package uk.co.stevewellington.shoppinglist

import uk.co.stevewellington.shoppinglist.infrasctructure.ItemStore
import uk.co.stevewellington.shoppinglist.model.ItemName

import scala.collection.mutable

class InMemoryItemStore extends ItemStore {

  val itemNames: mutable.HashSet[ItemName] = mutable.HashSet()

  override def createItemName(itemName: ItemName) = itemNames.add(itemName)

}
